pkg/monitortests/node/watchnodes/node.go

```diff
@@ -157,6 +148,52 @@ func startNodeMonitoring(ctx context.Context, m monitorapi.RecorderWriter, clien
 			}
 			return intervals
 		},
+		// This function is added to help detect unexpected
+		// node not ready.
+		// We want to fail the monitor test if a node goes not ready
+		// if it is unexpected.
+		// Unexpected in this case means that it went not ready outside
+		// of a MCO config update.
+		func(node, oldNode *corev1.Node) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+
+			oldConfig := oldNode.Annotations["machineconfiguration.openshift.io/currentConfig"]
+			newConfig := node.Annotations["machineconfiguration.openshift.io/currentConfig"]
+			isOldNodeReady := isNodeReady(oldNode)
+			isNewNodeReady := isNodeReady(node)
+			isConfigTheSame := oldConfig == newConfig
+
+			now := time.Now()
+			if (!isOldNodeReady || !isNewNodeReady) && isConfigTheSame {
```

Comment:
this doesn't seem quite right.  We only want a single interval created when we transition from Ready to NotReady when the configuration is the same.  In this case, every node update (say we change an image on the node status), will result in a new interval created, right?

perhaps `if isOldNodeReady && !isNewNodeReady && isConfigTheSame` ?

Reply:
I think we would want an interval in that case?

I thought that the goal would be to create an interval if a node goes NotReady outside of a MCO config? So I think that you would want this.

Reply:
I think your suggestion is good though. We only want an inteval is the newNode is NotReady? If NotReady -> Ready I guess we don't want an interval.